### PR TITLE
Chain Stuff And Naga and another mob render

### PR DESCRIPTION
**Pull request by ghost**
I readd render goblin and chainblock and throwable block entity
If you have any problems with this please let me know

### CODE REVIEW

**Comment by ghost** [comment]
I also Fixed many entity render stuff:

**Entity Render**:
Naga(Fully Fixed)
Lich Minion
LoyalZombie
Squirre
etc....

File: src/main/java/twilightforest/client/renderer/entity/RenderTFBlockGoblin.java [hunk 1]
```diff
@@ -1,34 +1,129 @@
...
+			Vec3d vec3d1 = this.getPosition(entity.block, (double) entity.block.getEyeHeight(), 1.0F);
+			return clippingHelper.isVisible(new AxisAlignedBB(vec3d1.x, vec3d1.y, vec3d1.z, vec3d.x, vec3d.y, vec3d.z));
+		}
+	}
+
+	private Vec3d getPosition(Entity p_177110_1_, double p_177110_2_, float p_177110_4_) {
+		double d0 = MathHelper.lerp((double) p_177110_4_, p_177110_1_.lastTickPosX, p_177110_1_.getX());
+		double d1 = MathHelper.lerp((double) p_177110_4_, p_177110_1_.lastTickPosY, p_177110_1_.getY()) + p_177110_2_;
+		double d2 = MathHelper.lerp((double) p_177110_4_, p_177110_1_.lastTickPosZ, p_177110_1_.getZ());
+		return new Vec3d(d0, d1, d2);
```
Comment (williewillus) [inline src/main/java/twilightforest/client/renderer/entity/RenderTFBlockGoblin.java:121]: are these methods copied from vanilla? please give the params proper names and add `// [VanillaCopy]` comments as appropriate.

File: src/main/java/twilightforest/client/renderer/entity/RenderTFChainBlock.java [hunk 2]
```diff
@@ -25,26 +34,62 @@ public RenderTFChainBlock(EntityRendererManager manager, Model modelTFSpikeBlock
...
 
-		//TODO: idk what's supposed to be done here
-//		renderManager.renderEntityStatic(chainBlock.chain1, partialTicks, false);
-//		renderManager.renderEntityStatic(chainBlock.chain2, partialTicks, false);
-//		renderManager.renderEntityStatic(chainBlock.chain3, partialTicks, false);
-//		renderManager.renderEntityStatic(chainBlock.chain4, partialTicks, false);
-//		renderManager.renderEntityStatic(chainBlock.chain5, partialTicks, false);
+	private void renderMultiBoundingBox(MatrixStack stack, IVertexBuilder builder, Entity entity, float p_229094_4_, float p_229094_5_, float p_229094_6_) {
+		AxisAlignedBB axisalignedbb = entity.getBoundingBox().offset(-entity.getX(), -entity.getY(), -entity.getZ());
+		WorldRenderer.drawBox(stack, builder, axisalignedbb, p_229094_4_, p_229094_5_, p_229094_6_, 1.0F);
```
Comment (williewillus) [inline src/main/java/twilightforest/client/renderer/entity/RenderTFChainBlock.java]: same here

File: src/main/java/twilightforest/client/renderer/entity/RenderTFNaga.java [hunk 3]
```diff
@@ -28,6 +39,65 @@ public void render(T entity, float entityYaw, float partialTicks, MatrixStack st
...
+			if (!partEntity.isInvisible()) {
+				renderSegment(entity, partEntity, entityYaw, partialTicks, stack, buffer, light);
+			}
+		}
+	}
+
+	@Override
+	protected void scale(T entity, MatrixStack stack, float p_225620_3_) {
+		super.scale(entity, stack, p_225620_3_);
+		//small size fix
```
Comment (williewillus) [inline src/main/java/twilightforest/client/renderer/entity/RenderTFNaga.java]: what is being "fixed"?

File: src/main/java/twilightforest/client/renderer/entity/RenderTFNaga.java [hunk 4]
```diff
@@ -28,6 +39,65 @@ public void render(T entity, float entityYaw, float partialTicks, MatrixStack st
...
+			float yawDiff = entity.rotationYaw - entity.prevRotationYaw;
+			if (yawDiff > 180) {
+				yawDiff -= 360;
+			} else if (yawDiff < -180) {
+				yawDiff += 360;
+			}
+			float yaw2 = entity.prevRotationYaw + yawDiff * partialTicks;
+
+			//GlStateManager.rotate(, 0.0F, 1.0F, 0.0F);
+			//GlStateManager.rotate(entity.rotationPitch, 1.0F, 0.0F, 0.0F);
```
Comment (williewillus) [inline src/main/java/twilightforest/client/renderer/entity/RenderTFNaga.java]: remove

File: src/main/java/twilightforest/client/renderer/entity/RenderTFSpikeBlock.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+		stack.multiply(Vector3f.POSITIVE_Z.getDegreesQuaternion(yaw));
+
+		this.model.render(stack, ivertexbuilder, light, OverlayTexture.DEFAULT_UV, 1.0F, 1.0F, 1.0F, 1.0F);
+		stack.pop();
+
+		//uh bruh...?
+		/*renderManager.render(goblin.block, x, y, z, yaw, partialTicks, stack, buffer, light);
+		renderManager.render(goblin.chain1, x, y, z, yaw, partialTicks, stack, buffer, light);
+		renderManager.render(goblin.chain2, x, y, z, yaw, partialTicks, stack, buffer, light);
+		renderManager.render(goblin.chain3, x, y, z, yaw, partialTicks, stack, buffer, light);*///renderEntity
```
Comment (williewillus) [inline src/main/java/twilightforest/client/renderer/entity/RenderTFSpikeBlock.java]: clean up commented code

File: src/main/java/twilightforest/world/TFWorld.java [hunk 6]
```diff
@@ -6,6 +6,7 @@
 import net.minecraft.block.Blocks;
 import net.minecraft.nbt.CompoundNBT;
 import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.GameRules;
```
Comment (williewillus) [inline src/main/java/twilightforest/world/TFWorld.java]: extraneous change?

**Review by williewillus — CHANGES_REQUESTED** [review]
looks pretty good, some cleanliness stuff

**Comment by williewillus** [comment]
Merge conflicts popped up

**Comment by ghost** [comment]
okey conflicts progrem fixed

**Pull request merged**